gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -277,13 +278,100 @@ impl GatewayLightning for ClnRpcService {
+            if let Some(outcome) = self
+                .interceptor
+                .lock()
+                .await
+                .outcomes
+                .lock()
+                .await
+                .remove(&hash)
+            {
+                // Translate action request into a cln rpc response for `htlc_accepted` event
+                let htlca_res = match action {
+                    Some(Action::Settle(Settle { preimage })) => {
+                        if let Ok(pk) = XOnlyPublicKey::from_slice(&preimage) {
+                            serde_json::json!({ "result": "resolve", "payment_key": pk.to_string() })
+                        } else {
+                            // 2002 failure code reports general temporary failure of this processing node.
+                            serde_json::json!({ "result": "fail", "failure_message": "2002" })
+                        }
+                    }
+                    Some(Action::Cancel(Cancel { reason: _ })) => {
+                        // TODO: Translate the reason into a BOLT 4 failure message
+                        // See: https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages
+                        serde_json::json!({ "result": "fail", "failure_message": "2002" })
+                    }
+                    None => {
+                        error!("No action specified for intercepted htlc id: {:?}", hash);
+                        responses.push(Err(Status::internal(
+                            "No action specified on this intercepted htlc",
+                        )));
+                        continue;
+                    }
+                };
+
+                // Send translated response to the HTLC interceptor for submission to the cln rpc
+                match outcome.send(htlca_res) {
+                    Ok(_) => {
+                        responses.push(Ok(CompleteHtlcsResponse {}));
+                    }
+                    Err(e) => {
+                        error!(
+                            "Failed to send htlc_accepted response to interceptor: {:?}",
+                            e
+                        );
+                        responses.push(Err(Status::internal(
+                            "Failed to send htlc_accepted outcome to interceptor",
+                        )));
+                    }
+                };
+                continue;
+            }
+
+            error!(
+                "No reference found for this intercepted and processed htlc id: {:?}",
+                intercepted_htlc_id
+            );
+            responses.push(Err(Status::internal(
+                "No reference found for this intercepted and processed htlc",
+            )))
+        }
+
+        let stream = async_stream::try_stream! {
+            for response in responses {
+                yield response?;
+            }
+        };
```

Comment:
I'm very sure there's a better way to get a stream from a `Vec`. E.g. `into_iter`+[`tokio_stream::iter`](https://docs.rs/tokio-stream/latest/tokio_stream/fn.iter.html)

Reply:
We need an iter `Stream`. Implements `poll_next`.

Instead of `async_stream` dependency, I ended up with a custom `impl Stream` for a `Vec` which is useful for now as we are not really parallel processing these requests.

Reply:
Generators might come in handy in future iterations